core/iwasm/interpreter/wasm_runtime.c

```diff
@@ -1126,6 +1126,17 @@ wasm_instantiate(WASMModule *module, bool is_sub_inst, uint32 stack_size,
     if (!module)
         return NULL;
 
+#if WASM_ENABLE_DEBUG_INTERP != 0
+    if (!is_sub_inst) {
+        os_mutex_lock(&module->ref_count_lock);
```

Comment:
Seems no need to add lock, but adding is also OK.

Reply:
There may be potential data race in this implementation, the ref_count is not increased before finishing instantiation, if some other threads instantiate the at the same time, the ref_count will be zero as the previous instantiation is not finished.

I've changed this code, we increase the ref_count here, and decrease it if instantiation failed